tests/e2e/test_async_client.py

```diff
@@ -168,6 +177,25 @@ async def test_invalid_job_id_error(self):
         )
         await dune.close_session()
 
+    async def test_disconnect(self):
+        dune = AsyncDuneClient(self.valid_api_key)
+        await dune.connect()
+        results = (await dune.refresh(self.query)).get_rows()
+        self.assertGreater(len(results), 0)
+        await dune.close_session()
+        self.assertTrue(cl._session.closed)
+
+    async def test_refresh_context_manager_singleton(self):
+        dune = AsyncDuneClient(self.valid_api_key)
+        async with dune as cl:
+            results = (await cl.refresh(self.query)).get_rows()
+        self.assertGreater(len(results), 0)
+
+    async def test_refresh_context_manager(self):
+        async with AsyncDuneClient(self.valid_api_key) as cl:
+            results = (await cl.refresh(self.query)).get_rows()
+        self.assertGreater(len(results), 0)
```

Comment:
So this is looking a bit more convenient. but as an end user all I want to have to do it
```py
(await dune.refresh(query)).get_rows()
```
Is it possible to make it so?

Reply:
Yeah, I tried it with the previous approach. It worked on my IPython but I think it fails when run inside a script (see your case). Not sure if we can make it the same as sync client. But I'll try to research it

Reply:
Once we have something that works a bit more conveniently, we can add some docs on how these things are used.

Reply:
Yes that... although I am not sure if that is the "right way" I am pretty new to async python and am hoping to learn a bit from you!

Reply:
Yeah, I'm learning about to_thread right now and benchmarking approach in order to come up with the best solution. One thing, could you please share the codes where you would like to use async function? This is needed to get a grasp of running environment and may be it would be easier to compare in real use-cases

Reply:
This open PR https://github.com/cowprotocol/dune-bridge/pull/37

Inside fetch/dune.py